src/Optimizely/Config/DatafileProjectConfig.php

```diff
@@ -230,19 +273,19 @@ public function __construct($datafile, $logger, $errorHandler)
 
         $this->_accountId = $config['accountId'];
         $this->_projectId = $config['projectId'];
-        $this->_anonymizeIP = isset($config['anonymizeIP'])? $config['anonymizeIP'] : false;
-        $this->_botFiltering = isset($config['botFiltering'])? $config['botFiltering'] : null;
+        $this->attributes = $config['attributes'] ?: [];
+        $this->audiences = $config['audiences'] ?: [];
+        $this->events = $config['events'] ?: [];
+        $this->typedAudiences = isset($config['typedAudiences']) ? $config['typedAudiences'] : [];
+        $this->_anonymizeIP = isset($config['anonymizeIP']) ? $config['anonymizeIP'] : false;
+        $this->_botFiltering = isset($config['botFiltering']) ? $config['botFiltering'] : null;
```

Comment:
what's the default value, why specifying explicit null.

Reply:
default value for botFiltering was not mentioned anywhere i did not change this line it is as it was.